app/js/containers/ArtifactGenerator.js

```diff
@@ -24,7 +24,7 @@ const mapDispatchToProps = dispatch => ({
         const fd = new FormData(e.target);
         const data = {};
         for (const [key, value] of fd) {
-            if (value === '' || value === undefined) {
+            if (key !== 'source_format' && (value === '' || value === undefined)) {
```

Comment:
So it was important that the other values besides `source_format` be passed as `''` if they are not entered in order to trigger `@fail_elegantly` in the backend python code, so I ended up modifying this part of the code to first check if the key is `source_format` and the value is `''`, in which case we set the value to null, and then to check if that is not the case if value is undefined etc, and then finally to set the value as was done before.  This should make the code more readable here (and maybe a bit less hack-y looking) and also fix the problem you pointed out in `workspace.py` with `source_format` not being null by default.

Let me know if you don't like this style and I am happy to change it.  Thanks!